internal/ingress/controller/parser/kongstate/route.go

```diff
@@ -303,4 +305,54 @@ func (r *Route) overrideByKongIngress(log logrus.FieldLogger, kongIngress *confi
 		}
 		r.SNIs = SNIs
 	}
+	if ir.RequestBuffering != nil {
+		r.RequestBuffering = kong.Bool(*ir.RequestBuffering)
+	}
+	if ir.ResponseBuffering != nil {
+		r.ResponseBuffering = kong.Bool(*ir.ResponseBuffering)
+	}
+}
+
+// overrideRequestBuffering ensures defaults for the request_buffering option
+func (r *Route) overrideRequestBuffering(log logrus.FieldLogger, anns map[string]string) {
+	// by default request_buffering should be enabled. It only needs to be
+	// disabled under special use cases, including large payloads.
+	r.RequestBuffering = kong.Bool(true)
+
+	annotationValue, ok := annotations.ExtractRequestBuffering(anns)
+	if !ok {
+		// the annotation is not set, default to enabled
+		return
+	}
+
+	isEnabled, err := strconv.ParseBool(annotationValue)
+	if err != nil {
+		// the value provided is not a parseable boolean
+		log.WithField("kongroute", r.Name).Errorf("invalid request_buffering value: %w", err)
+		return
+	}
+
+	r.RequestBuffering = kong.Bool(isEnabled)
+}
+
+// overrideResponseBuffering ensures defaults for the response_buffering option
+func (r *Route) overrideResponseBuffering(log logrus.FieldLogger, anns map[string]string) {
+	// by default response_buffering should be enabled. It only needs to be
+	// disabled under special use cases, including large payloads.
+	r.ResponseBuffering = kong.Bool(true)
```

Comment:
Ditto request buffering override comment. No controller default here.

Reply:
Sorry, I had already realized this and had been changing this before you posted this!